pennylane/tape/operation_recorder.py

```diff
@@ -15,12 +15,12 @@
 This module contains the :class:`OperationRecorder`.
 """
 # pylint: disable=too-many-arguments
-from pennylane.queuing import QueuingManager
+from pennylane.queuing import AnnotatedQueue, QueuingManager, process_queue
 
-from .tape import QuantumTape, QuantumScript
+from .tape import QuantumScript
 
 
-class OperationRecorder(QuantumTape):
+class OperationRecorder(QuantumScript, AnnotatedQueue):

```

Comment:
this file contains the only changes to `pennylane/` and is probably worthy of your attention. As I state in the PR description, I just took the relevant code from `QuantumTape` and put it here.

Reply:
Thank you for specifying that! It will help for the review

Reply:
@timmysilv not sure if this is a breaking change or not, but the operation recorder is used more than we realize externally - last time we tried to deprecate/remove it (@rmoyard I think you might have been involved here?) we ended up having to add it back to the codebase as a legacy.

Reply:
Yes I think we had some issues, with some templates for example that were using it but now it could potentially be different as we have quantum script for treating a list of operations directly.

Reply:
We should check where it is still used and how.

Reply:
Yes, and if we have a replacement for it. But even though we don't use the operation recorder internally, we should avoid breaking it or its behaviour until we can communicate a replacement.

Reply:
the only change between here and `QuantumTape` is that I removed integrated support for nested OperationRecorder instances (QuantumScript also doesn't natively support nested recording). I left the `do_queue` keyword in the constructor, but it is unused and ignored. There is no `QuantumTape._lock.acquire()` done either.